sgkit/stats/grm.py

```diff
@@ -147,3 +149,655 @@ def genomic_relationship(
+        output variables into a single dataset, otherwise return only
+        the computed output variables.
+        See :ref:`dataset_merge` for more details.
+
+    Returns
+    -------
+    A dataset containing :data:`sgkit.variables.stat_inverse_relationship_spec`
+    The dimensions are named ``samples_0`` and ``samples_1``. If a subset of
+    samples was specified, then nan values are used to indicate relationships
+    that are not included within the subset.
+
+    Examples
+    --------
+
+    >>> import xarray as xr
+    >>> import sgkit as sg
+    >>> ds = xr.Dataset()
+    >>> ds["stat_pedigree_relationship"] = ["samples_0", "samples_1"], [
+    ...     [1.   , 0.   , 0.   , 0.5  , 0.   , 0.25 ],
+    ...     [0.   , 1.   , 0.   , 0.5  , 0.5  , 0.5  ],
+    ...     [0.   , 0.   , 1.   , 0.   , 0.5  , 0.25 ],
+    ...     [0.5  , 0.5  , 0.   , 1.   , 0.25 , 0.625],
+    ...     [0.   , 0.5  , 0.5  , 0.25 , 1.   , 0.625],
+    ...     [0.25 , 0.5  , 0.25 , 0.625, 0.625, 1.125]
+    ... ]
+    >>> sg.invert_relationship_matrix(
+    ...     ds,
+    ...     relationship="stat_pedigree_relationship",
+    ... ).stat_inverse_relationship.values  # doctest: +NORMALIZE_WHITESPACE
+    array([[ 1.5,  0.5,  0. , -1. ,  0. ,  0. ],
+           [ 0.5,  2. ,  0.5, -1. , -1. ,  0. ],
+           [ 0. ,  0.5,  1.5,  0. , -1. ,  0. ],
+           [-1. , -1. ,  0. ,  2.5,  0.5, -1. ],
+           [ 0. , -1. , -1. ,  0.5,  2.5, -1. ],
+           [ 0. ,  0. ,  0. , -1. , -1. ,  2. ]])
+    >>> # inverse of a sub-matrix
+    >>> ds["subset_sample"] = "samples", [False, False, False, True, True, True]
+    >>> sg.invert_relationship_matrix(
+    ...     ds,
+    ...     relationship="stat_pedigree_relationship",
+    ...     subset_sample="subset_sample",
+    ... ).stat_inverse_relationship.values.round(3)  # doctest: +NORMALIZE_WHITESPACE
+    array([[   nan,    nan,    nan,    nan,    nan,    nan],
+           [   nan,    nan,    nan,    nan,    nan,    nan],
+           [   nan,    nan,    nan,    nan,    nan,    nan],
+           [   nan,    nan,    nan,  1.567,  0.233, -1.   ],
+           [   nan,    nan,    nan,  0.233,  1.567, -1.   ],
+           [   nan,    nan,    nan, -1.   , -1.   ,  2.   ]])
+    """
+    A = da.array(ds[relationship].data)
+    if subset_sample is None:
+        Ainv = da.linalg.inv(A)
+    else:
+        idx = ds[subset_sample].values
+        Ainv = _sub_matrix_inv(A, idx, chunks=subset_rechunk)
+    new_ds = create_dataset(
+        {
+            variables.stat_inverse_relationship: (
+                ("samples_0", "samples_1"),
+                Ainv,
+            )
+        }
+    )
+    return conditional_merge_datasets(ds, new_ds, merge)
+
+
+def _hybrid_full_matrix_Martini(
+    Ainv: ArrayLike, G: ArrayLike, tau: float = 1, omega: float = 1
+):
+    # calculate sub-matrices of H
+    if (tau == 1) and (omega == 1):
+        H = G
+    elif omega == 1:
+        H = 1 / tau * G
```

Comment:
Maybe comment here on why we're not using the Dask version (or delete the Dask line)?

Reply:
The commented code was meant to be deleted. The lines are equivalent, but I originally had the less efficient version.